filters/diag/diag.go

```diff
@@ -182,15 +181,15 @@ func NewNormalResponseLatency() filters.Spec { return &jitter{typ: normalRespons
 func (r *random) Name() string { return RandomName }
 
 func (r *random) CreateFilter(args []interface{}) (filters.Filter, error) {
-	if len(args) != 1 {
-		return nil, filters.ErrInvalidFilterParameters
+	a := filters.Args(args)
+	len, err := a.Int64(), a.Err()
+	if err != nil {
+		return nil, err
 	}
-
-	if l, ok := args[0].(float64); ok {
-		return &random{rand: rand.New(rand.NewSource(time.Now().UnixNano())), len: int64(l)}, nil // #nosec
-	} else {
+	if len < 0 {
```

Comment:
Arguments could be assigned to new variables and then addition validation performed.